skbeam/core/accumulators/binned_statistic.py

```diff
@@ -70,8 +70,13 @@ def __init__(self, sample, statistic='mean',
             A sequence of lower and upper bin edges to be used if the
             edges are not given explicitely in `bins`. Defaults to the
             minimum and maximum values along each dimension.
-
+        mask : array_like
+            array of ones and zeros with the same shape as `sample`.
+            Values with mask==0 will be ignored.
         """
+        if mask is None:
+            mask = np.ones_like(sample)
```

Comment:
Hi Julien,

I think you're correct that my mask shape was wrong.  If sample has shape (N,D) then I think mask should have shape (N).  e.g. if one is producing a projection onto 2D r-phi, the "bad pixel" mask should be the same for both r and phi.  I was testing the 1D mask (with help from Tom) by comparing with the output with scipy's binned_statistic, but I wasn't testing the 2D case.  I will try to add that, comparing with binned_statistic_2d.

Thanks for all the ideas.  Appreciated...